flow/project.py

```diff
@@ -868,11 +1043,151 @@ def add_operation(self, name, operation, directives=None):
             The operation specific directives.
         :type directives:
             dict
+        :param flow_aggregate:
+            The operation specific aggregate parameters specified
+            using the :py:class:`aggregate`
+        :type flow_aggregate:
+            tuple
+        :param flow_select:
+            The operation specific filter parameters specified
+            using the :py:class:`select`
+        :type flow_select:
+            callable
         """
         self.operations[name] = operation
         if directives is not None:
             self.operation_directives[name] = directives
 
+        if self.flow_aggregate == (None, None):
+            self.flow_aggregate = flow_aggregate
+            self.aggregate_per_group = False
+        elif not self.aggregate_per_group:
```

Comment:
Is this logic necessary? An operation could theoretically support multiple modes of aggregation/selecting. We should just enforce the aggregation and selecting the group was registered with. If it isn't the same as the singleton group, then that is a problem with the user's script.